core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -579,6 +579,7 @@ boolean isPreserveAliases() {
 
         private Set<String> innerResolve(Context context, List<String> expressions, IndicesOptions options, MetaData metaData) {
             Set<String> result = null;
+            boolean wildcardSeen = false;
             for (int i = 0; i < expressions.size(); i++) {
                 String expression = expressions.get(i);
                 if (aliasOrIndexExists(metaData, expression)) {
```

Comment:
I don't know why it doesn't initialize the `result` set like the other null checks do, but I do think *all* of this code should be cleaned up after this (I hesitate to touch more code as this is supposed to be backported all the way back to 2.4)